### PR TITLE
Add mhp dataset for both bottomup and topdown; Correct val pipline.

### PR DESCRIPTION
**Pull request by dreamPoet**
There is a format bug caused by the conflict between isort and yapf.

### CODE REVIEW

File: configs/bottom_up/hrnet/coco/hrnet_w48_coco_512x512.py [hunk 1]
```diff
@@ -172,7 +172,7 @@
 
 data_root = 'data/coco'
 data = dict(
-    samples_per_gpu=16,
+    samples_per_gpu=8,
```
Comment (jin-s13) [inline configs/bottom_up/hrnet/coco/hrnet_w48_coco_512x512.py:175]: no need to change this

File: configs/top_down/darkpose/coco/res101_coco_256x192_dark.py [hunk 2]
```diff
@@ -96,7 +96,7 @@
         ]),
 ]
 
-valid_pipeline = [
+val_pipeline = [
```
Comment (innerlee) [inline configs/top_down/darkpose/coco/res101_coco_256x192_dark.py:99]: did you searched through the whole codebase for `valid_pipeline`?

Comment (jin-s13) [reply configs/top_down/darkpose/coco/res101_coco_256x192_dark.py:99]: Yes.

File: configs/top_down/resnet/README.md [hunk 3]
```diff
@@ -61,7 +61,7 @@ Following the common setting, the models are trained on COCO train dataset, and
 
 | Arch  | Input Size | AP | AP<sup>50</sup> | AP<sup>75</sup> | AR | AR<sup>50</sup> | ckpt | log |
 | :-------------- | :-----------: | :------: | :------: | :------: | :------: | :------: |:------: |:------: |
-| [pose_resnet_101](/configs/top_down/resnet/mhp/res50_mhp_256x192.py) | 256x192 | 0.583 | 0.897 | 0.669 | 0.636 | 0.918 | [ckpt]| [log]|
+| [pose_resnet_101](/configs/top_down/resnet/mhp/res50_mhp_256x192.py) | 256x192 | 0.583 | 0.897 | 0.669 | 0.636 | 0.918 | [ckpt](https://openmmlab.oss-cn-hangzhou.aliyuncs.com/mmpose/top_down/resnet/res50_mhp_256x192-28c5b818_20201229.pth) | [log](https://openmmlab.oss-cn-hangzhou.aliyuncs.com/mmpose/top_down/resnet/res50_mhp_256x192_20201229.log.json) |
```
Comment (innerlee) [inline configs/top_down/resnet/README.md:64]: check the url format

File: README.md [hunk 4]
```diff
@@ -90,12 +90,14 @@ Supported datasets:
...
 - [x] [CrowdPose](https://github.com/Jeff-sjtu/CrowdPose)
+- [x] [PoseTrack18](https://posetrack.net/users/download.php)
+- [x] [MHP](https://lv-mhp.github.io/dataset)
 - [x] [sub-JHMDB](http://jhmdb.is.tue.mpg.de/dataset)
-- [x] [H36m](http://vision.imar.ro/human3.6m/description.php)
 - [x] [OneHand10K](https://www.yangangwang.com/papers/WANG-MCC-2018-10.html)
 - [x] [FreiHand](https://lmb.informatik.uni-freiburg.de/projects/freihand/)
 - [x] [CMU Panoptic HandDB](http://domedb.perception.cs.cmu.edu/handdb.html)
 - [x] [InterHand2.6M](https://mks0601.github.io/InterHand2.6M/)
+- [x] [H36m](http://vision.imar.ro/human3.6m/description.php)
```
Comment (innerlee) [inline README.md:100]: `H36m` -> `H36M`

File: mmpose/datasets/datasets/bottom_up/bottom_up_mhp.py [hunk 5]
```diff
@@ -0,0 +1,374 @@
...
+                if len(self.coco.getAnnIds(imgIds=img_id, iscrowd=None)) > 0
+            ]
+        self.num_images = len(self.img_ids)
+        self.id2name, self.name2id = self._get_mapping_id_name(self.coco.imgs)
+        self.dataset_name = 'mhp'
+
+        print(f'=> num_images: {self.num_images}')
+
+    @staticmethod
+    def _get_mapping_id_name(imgs):
```
Comment (innerlee) [inline mmpose/datasets/datasets/bottom_up/bottom_up_mhp.py:106]: put this into `BottomUpBaseDataset`?

File: mmpose/datasets/datasets/bottom_up/bottom_up_mhp.py [hunk 6]
```diff
@@ -0,0 +1,374 @@
...
+
+        kpts = defaultdict(list)
+        # iterate over images
+        for idx, _preds in enumerate(preds):
+            str_image_path = image_paths[idx]
+            image_id = self.name2id[os.path.basename(str_image_path)]
+            # iterate over people
+            for idx_person, kpt in enumerate(_preds):
+                # use bbox area
+                area = (np.max(kpt[:, 0]) - np.min(kpt[:, 0])) * (
```
Comment (innerlee) [inline mmpose/datasets/datasets/bottom_up/bottom_up_mhp.py]: what's the shape of `kpt`?

File: mmpose/datasets/datasets/bottom_up/bottom_up_mhp.py [hunk 7]
```diff
@@ -0,0 +1,374 @@
...
+                    'image_id':
+                    img_kpt['image_id'],
+                    'category_id':
+                    cat_id,
+                    'keypoints':
+                    list(key_point),
+                    'score':
+                    img_kpt['score'],
+                    'bbox':
+                    list([left_top[0], left_top[1], w, h])
```
Comment (innerlee) [inline mmpose/datasets/datasets/bottom_up/bottom_up_mhp.py]: this line is too long

File: mmpose/datasets/datasets/bottom_up/bottom_up_mhp.py (hunk 6 above)
Comment (jin-s13) [reply mmpose/datasets/datasets/bottom_up/bottom_up_mhp.py]: [K, 3+tag_num]

tag_num >= 1

File: mmpose/datasets/datasets/bottom_up/bottom_up_mhp.py (hunk 7 above)
Comment (innerlee) [reply mmpose/datasets/datasets/bottom_up/bottom_up_mhp.py]: redundant `list()` call

**Review by innerlee — APPROVED** [review]

**Pull request merged**